src/Kiota.Builder/SearchProviders/GitHub/Authentication/TempFolderStorageService.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Kiota.Builder.SearchProviders.GitHub.Authentication;
+
+public class TempFolderTokenStorageService : ITokenStorageService {
+    public required ILogger Logger { get; init;}
+    public required string FileName { get; init; }
+    private string GetTokenCacheFilePath() => Path.Combine(Path.GetTempPath(), "kiota", "auth", $"{FileName}.txt");
+    public async Task<string> GetTokenAsync(CancellationToken cancellationToken)
+    {
+        try {
+            var target = GetTokenCacheFilePath();
+            if(!await IsTokenPresentAsync(cancellationToken))
+                return null;
+            var result = await File.ReadAllTextAsync(target, cancellationToken).ConfigureAwait(false);
+            return result;
+        } catch (Exception ex) {
+            Logger.LogWarning(ex, "Error while reading token from cache.");
+            return null;
+        }
+    }
+
+    public async Task SetTokenAsync(string value, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var target = GetTokenCacheFilePath();
+            var directory = Path.GetDirectoryName(target);
+            if (!Directory.Exists(directory))
+                Directory.CreateDirectory(directory);
+            await File.WriteAllTextAsync(target, value, cancellationToken).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            Logger.LogWarning(ex, "Error while writing token to cache.");
+        }
+    }
+
+    public async Task<bool> IsTokenPresentAsync(CancellationToken cancellationToken)
+    {
+        try {
+            var target = GetTokenCacheFilePath();
+            if (!File.Exists(target))
+                return false;
+            var fileDate = File.GetLastWriteTime(target);
+            if(fileDate > DateTime.Now.AddMonths(6)) {
```

Comment:
I suspect we may need to re-evaluate this check. It looks like this will always evaluate to false as the last write time can't be 6 months in the future.

Reply:
```suggestion
            if(fileDate > DateTime.Now.AddMonths(-6)) {
```
Thanks, that's what I meant